<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <comment>Copyright (C) 2012-2015 Robert Ch√©ramy (robert@cheramy.net)

This file is part of YASW (Yet Another Scan Wizard).

YASW is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

YASW is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with YASW.  If not, see http://www.gnu.org/licenses/.
</comment>
 <class>ThresholdWidget</class>
 <widget class="QWidget" name="ThresholdWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>706</width>
    <height>556</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QCheckBox" name="enable">
       <property name="text">
        <string>Enable Filter</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="preview">
       <property name="text">
        <string>Preview</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>293</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="1" column="0" rowspan="9">
    <widget class="BaseFilterGraphicsView" name="view">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="2">
    <widget class="QCheckBox" name="preprocessNoise">
     <property name="text">
      <string>Preprocessing: bilatreal filter for noise reduction</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QRadioButton" name="rbThreshold_2">
     <property name="text">
      <string>No thresholding</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QRadioButton" name="rbOtsuThreshold">
     <property name="text">
      <string>Otsu thresholding</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="2">
    <widget class="QFrame" name="frame_3">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="labelWhite_3">
        <property name="text">
         <string>calculated threshold </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="otsuThreshold">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QRadioButton" name="rbThreshold">
     <property name="text">
      <string>Normal threshold</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1" colspan="2">
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelWhite_2">
        <property name="text">
         <string>threshold</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="threshold">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>125</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QRadioButton" name="rbAdaptativeThreshold">
     <property name="text">
      <string>Adaptative threshold</string>
     </property>
    </widget>
   </item>
   <item row="8" column="1" colspan="2">
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="labelBlack">
        <property name="text">
         <string>C</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="cval">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelBlack_2">
        <property name="text">
         <string>BlockSize</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="blocksize">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>100</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="9" column="2">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>150</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BaseFilterGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>basefiltergraphicsview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>cval</sender>
   <signal>editingFinished()</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>686</x>
     <y>294</y>
    </hint>
    <hint type="destinationlabel">
     <x>387</x>
     <y>382</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blocksize</sender>
   <signal>editingFinished()</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>686</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>460</x>
     <y>390</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>threshold</sender>
   <signal>editingFinished()</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>686</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>457</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preprocessNoise</sender>
   <signal>toggled(bool)</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>582</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbThreshold</sender>
   <signal>toggled(bool)</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>587</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbAdaptativeThreshold</sender>
   <signal>toggled(bool)</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>587</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbOtsuThreshold</sender>
   <signal>toggled(bool)</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>449</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>683</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbThreshold_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>ThresholdWidget</receiver>
   <slot>paramChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>656</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>paramChanged()</slot>
 </slots>
</ui>
